mongo/datadog_checks/mongo/mongo.py

```diff
@@ -163,12 +177,14 @@ def _build_metric_list_to_collect(self):
 
     def check(self, _):
         try:
-            api = MongoApi(self._config, self.log)
-            self.log.debug("Connected!")
+            self._check()
         except Exception:
-            self.service_check(SERVICE_CHECK_NAME, AgentCheck.CRITICAL, tags=self._config.service_check_tags)
+            self._api_client = None
```

Comment:
Would it be safer to only clear the _api_client if `ConnectionFailure` is thrown? 

The reason for the initial request to change this was to avoid losing visibility if we hit the max number of connections on mongo, so if we proactively ditch the client here then I'm thinking there's a bigger chance of dropping our connection unnecessarily.

Somewhat related, does this not leak connections? If so, I guess it's not something that's introduced by this PR, but I didn't see anywhere (also in the old code) where we're explicitly closing connections (although I guess they get closed when the underlying object gets collected?)